### ISSUE TITLE
[Bug] KeyVaultOperation support refreshing only for the last one

### ISSUE THREAD

**Issue by chenrujun** [Azure/azure-sdk-for-java#26356]
### Context

Here is the current code in KeyVaultOperation:
```java

   private static Timer timer;

   public KeyVaultOperation(final SecretClient secretClient,
                             final Duration refreshDuration,
                             List<String> secretKeys,
                             boolean caseSensitive) {
        ......
        if (refreshInMillis > 0) {
            synchronized (KeyVaultOperation.class) {
                if (timer != null) {
                    try {
                        timer.cancel();
                        timer.purge();
                    } catch (RuntimeException runtimeException) {
                        LOGGER.error("Error of terminating Timer", runtimeException);
                    }
                }
                timer = new Timer(true);
                final TimerTask task = new TimerTask() {
                    @Override
                    public void run() {
                        refreshProperties();
                    }
                };
                timer.scheduleAtFixedRate(task, refreshInMillis, refreshInMillis);
            }
        }
    }

```

### Problem

Seems it only last KeyVaultOperation supports auto refresh.

### Goal

1. Confirm whether only the last KeyVaultOperation supports auto refresh.
2. If only the last one supports auto refresh, it's a bug, we need to fix it.

**Comment by moarychan** [comment]
I think this issue exactly exists. Only the last keyvault properties source timer works, the previous timer task will be overridden the last one.

![image](https://user-images.githubusercontent.com/19926623/156323853-b58cc034-1b6c-450a-8ff7-40222e58bc09.png)

**Comment by moarychan** [comment]
https://github.com/Azure/azure-sdk-for-java/pull/41703

**Comment by moarychan** [comment]
The fixes will be available in Spring Cloud Azure 5.1.6.0.
